src/dbnode/client/host_queue.go

```diff
@@ -229,45 +246,101 @@ func (q *queue) drain() {
 			switch v := ops[i].(type) {
 			case *writeOperation:
 				namespace := v.namespace
-				idx := currWriteOpsByNamespace.indexOf(namespace)
-				if idx == -1 {
-					value := namespaceWriteBatchOps{
-						namespace:                            namespace,
-						opsArrayPool:                         q.opsArrayPool,
-						writeBatchRawRequestElementArrayPool: q.writeBatchRawRequestElementArrayPool,
+				if q.serverSupportsV2APIs {
```

Comment:
This function looks super large. Any way to factor it out into smaller functions here?